### DISCUSSION TITLE
Setting command tags

### DISCUSSION THREAD

**Discussion by StephanusVictor** [riclolsen/json-scada#68] [Q&A] [answered]
Hi Ricardo,

How do I set a command tag when an object (“button”) is clicked on a svg screen? For instance, for implementing a process start button.

Thanks for this project!

**Comment by riclolsen** [comment, marked as answer]
Thank you very much for liking the project, Stephanus.
The function $W.directCommandExec(point_key_number, value) can be used to execute commands from scripts (no confirmation will be required) that can be linked to events (like a mouse click), using the SVG Editor (object properties|script|mousedown). The first parameter is the numeric point key of the command. Value parameter is the value of the command (can be a number for analog commands or “ON” and “OFF” for digital commands.
Best regards,
Ricardo

**Reply by StephanusVictor** [reply]
Thanks for the help Ricardo,

I am getting the following error in my browser console when i click the "button"

websage.js:1326 
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'getElementById') at websage.js:1326:36
(anonymous) @ websage.js:1326
Promise.catch (async)
directCommandExec @ websage.js:1324
onmouseup @ display.html?SVGFILE=../svg/arniston.svg:265

The tag is setup with as a digital command
Mouseup script:   $W.directCommandExec(20010084, 1); 


Kind regards
Stephanus

**Reply by riclolsen** [reply]
There is a problem indeed, there is attempt to write feedback to a window that doesn't exist. However I think the command should be delivered to the server before the error, so it is supposed to work anyway. I'll post a fix for that.
Thank you very much for spotting the problem.

**Reply by riclolsen** [reply]
Fixed in the repo.

Replace this file C:\json-scada\src\htdocs\websage.js

with this updated code

https://github.com/riclolsen/json-scada/blob/33d2abb4d4392b3e5df63b310e021dd58a1c5964/src/htdocs/websage.js
